pkg/distributor/distributor.go

```diff
@@ -647,6 +655,10 @@ func (d *Distributor) PushWithCleanup(ctx context.Context, req *mimirpb.WriteReq
 		return nil, httpgrpc.Errorf(http.StatusTooManyRequests, validation.NewRequestRateLimitedError(d.limits.RequestRate(userID), d.limits.RequestBurstSize(userID)).Error())
 	}
 
+	if d.cfg.InstanceLimits.MaxInflightPushRequestsTotalSize > 0 && inflightSize > d.cfg.InstanceLimits.MaxInflightPushRequestsTotalSize {
```

Comment:
Seems like this should go above the rate limit request, with the other instance limit checks